src/fleet/azext_fleet/custom.py

```diff
@@ -143,22 +144,21 @@ def update_fleet(cmd,
         operation_group="fleets"
     )
 
-    if enable_managed_identity is None:
-        managed_service_identity = None
-        if assign_identity is not None:
-            raise CLIError("Cannot assign identity without enabling managed identity.")
-    elif enable_managed_identity is False:
-        managed_service_identity = fleet_managed_service_identity_model(type="None")
-    else:
+    managed_service_identity = None  # Do not patch by default
+    if mi_system_assigned is True and mi_user_assigned is not None:
+        raise CLIError("A fleet can either have a system or user assigned identity, not both.")
+
+    if mi_system_assigned is True:
         managed_service_identity = fleet_managed_service_identity_model(type="SystemAssigned")
-        if assign_identity is not None:
-            user_assigned_identity_model = cmd.get_models(
-                "UserAssignedIdentity",
-                resource_type=CUSTOM_MGMT_FLEET,
-                operation_group="fleets"
-            )
-            managed_service_identity.type = "UserAssigned"
-            managed_service_identity.user_assigned_identities = {assign_identity: user_assigned_identity_model()}
+    elif mi_user_assigned is not None:
+        managed_service_identity = fleet_managed_service_identity_model(type="UserAssigned")
+        user_assigned_identity_model = cmd.get_models(
+            "UserAssignedIdentity",
+            resource_type=CUSTOM_MGMT_FLEET,
+            operation_group="fleets")
+        managed_service_identity.user_assigned_identities = {mi_user_assigned: user_assigned_identity_model()}
+    else:
+        managed_service_identity = fleet_managed_service_identity_model(type="None")
```

Comment:
Won't this remove by default, despite the comment above?

Why does the update code use if / elif and the create code use if / if ?  Be consistent.

Reply:
reverted back to elseif.